### ISSUE TITLE
Conflicting return type in implementation of supportedInterfaceOrientations

### ISSUE THREAD

**Issue by AaoIi** [John-Lluch/SWRevealViewController#474]
i am having a warning in Xcode 7 in the following method : 
- (NSUInteger)supportedInterfaceOrientations
  {
  // we could have simply not implemented this, but we choose to call super to make explicit that we
  // want the default behavior.
  return [super supportedInterfaceOrientations];
  }

Conflicting return type in implementation of supportedInterfaceOrientations: UIinterfaceOrientationMask (aka enum UIinterfaceOrientationMask vs NSUIinteger (aka unsigned int)).

i hope it get fixed :)

**Comment by ZsoltMolnarrr** [comment]
I am trying to use SWRVC in my modular Swift project. I added it to an embedded (ObjC) Framework created for 3rd Party ObjC UI libraries and this warning appeared for me too.

**Comment by AaoIi** [comment]
I guess i fixed the warning by changing NSUInteger to UIInterfaceOrientationMask, i don't know but it seems to be working okay: 
- (UIInterfaceOrientationMask)supportedInterfaceOrientations
  {
  // we could have simply not implemented this, but we choose to call super to make explicit that we
  // want the default behavior.
  return [super supportedInterfaceOrientations];
  }

**Comment by bryamrr** [comment]
@AaoIi That works fine! Thx

**Comment by Souf-R** [comment]
Hi,

I hope this tweak (**_for iOS 9+_**) helps. Go to `SWRevealViewController.m` and replace this method:

``` swift
- (NSUInteger)supportedInterfaceOrientations
{
    // we could have simply not implemented this, but we choose to call super to make explicit that we
    // want the default behavior.
    return [super supportedInterfaceOrientations];
}
```

by this one:

``` swift
#if __IPHONE_OS_VERSION_MAX_ALLOWED < 90000
- (NSUInteger)supportedInterfaceOrientations
#else
- (UIInterfaceOrientationMask)supportedInterfaceOrientations
#endif
{
    // we could have simply not implemented this, but we choose to call super to make explicit that we
    // want the default behavior.
    return [super supportedInterfaceOrientations];
}
```

**Comment by phoney** [comment]
I'm still seeing this warning in v2.4.0. Needless to say it should be rolled into the latest version.

**Comment by iDevelopper** [comment]
No, he said to change the source code of SWRevealViewController.m. But this warning is not very important as UIInterfaceOrientationMask is an NSUInteger. See the documentation.
